src/Components/SimpleTensorNetwork.jl

```diff
@@ -216,7 +216,11 @@ function rmtensor!(tn::SimpleTensorNetwork, tensor::Tensor)
     # remove indices if they were removed
     # TODO maybe we should refactor `rmtensor!` to check if we use a `Network` underneath and then, use the `RemoveVertexEffect` and `RemoveEdgeEffect` effects?
     for edge in edge_set
-        if !hasedge(tn, edge)
+        # Delete edge if it is not incident to any other tensor
+        vertex_set = edge_incidents(tn, edge)
+
+        if isempty(vertex_set)
+            rmedge!(tn.network, edge)
```

Comment:
This should be the default behavior of the `PruneEdges <: EdgePersistence` trait, but it seems like I left out sth in the implementation of `IncidentNetwork` that `SimpleTensorNetwork` uses.

Reply:
mmm not sure, but they way you're fixing it (removing the edges from `tn.network`) is literally something that `IncidentNetwork` should already manage. like the code you're touching is purely from Networks.jl